validator/db/manage.go

```
@@ -0,0 +1,137 @@
+				}
+				allProposals = append(allProposals, *pubKeyProposals)
+				return nil
+			}); err != nil {
+				return errors.Wrapf(err, "Could not retrieve proposals for database in %s", store.databasePath)
+			}
+
+			attestationsBucket := tx.Bucket(historicAttestationsBucket)
+			if err := attestationsBucket.ForEach(func(pubKey, v []byte) error {
+				attestations := pubKeyAttestations{
+					PubKey:       make([]byte, len(pubKey)),
+					Attestations: make([]byte, len(v)),
+				}
+				copy(attestations.PubKey, pubKey)
+				copy(attestations.Attestations, v)
+				allAttestations = append(allAttestations, attestations)
+				return nil
+			}); err != nil {
+				return errors.Wrapf(err, "Could not retrieve attestations for database in %s", store.databasePath)
+			}
+
+			return nil
+		}); err != nil {
+			return err
+		}
+	}
+
+	if err := createTargetStore(targetDirectory, allProposals, allAttestations); err != nil {
+		return errors.Wrapf(err, "Could not create target store")
+	}
+
+	return nil
+}
+
+func getPubKeyProposals(pubKey []byte, proposalsBucket *bolt.Bucket) (*pubKeyProposals, error) {
+	pubKeyProposals := pubKeyProposals{
+		PubKey:    pubKey,
+		Proposals: []epochProposals{},
+	}
+
+	pubKeyBucket := proposalsBucket.Bucket(pubKey)
+	if err := pubKeyBucket.ForEach(func(epoch, v []byte) error {
+		epochProposals := epochProposals{
+			Epoch:     make([]byte, len(epoch)),
+			Proposals: make([]byte, len(v)),
+		}
+		copy(epochProposals.Epoch, epoch)
+		copy(epochProposals.Proposals, v)
+		pubKeyProposals.Proposals = append(pubKeyProposals.Proposals, epochProposals)
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+
+	return &pubKeyProposals, nil
+}
+
+func createTargetStore(
+	targetDirectory string,
+	allProposals []pubKeyProposals,
+	allAttestations []pubKeyAttestations) error {
+
+	newStore, err := NewKVStore(targetDirectory)
+	defer func() {
+		if e := newStore.Close(); e != nil {
+			err = errors.Wrap(err, "Could not close the merged database")
```

again this does nothing

---

By "nothing" you mean that the errors are not handled, right? The code does close the db connections.

---

Yeap, you are defining the errors but doing nothing with them. You should log an error here.